src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -264,18 +266,42 @@ public void SetAll(bool value)
         ** Exceptions: ArgumentException if value == null or
         **             value.Length != this.Length.
         =========================================================================*/
-        public BitArray And(BitArray value)
+        public unsafe BitArray And(BitArray value)
         {
             if (value == null)
                 throw new ArgumentNullException(nameof(value));
             if (Length != value.Length)
                 throw new ArgumentException(SR.Arg_ArrayLengthsDiffer);
 
-            for (int i = 0; i < m_array.Length; i++)
+            int count = m_array.Length;
+
+            switch (count)
             {
-                m_array[i] &= value.m_array[i];
+                case 3: m_array[2] &= value.m_array[2]; goto case 2;
+                case 2: m_array[1] &= value.m_array[1]; goto case 1;
+                case 1: m_array[0] &= value.m_array[0]; goto _done;
+                case 0: goto _done;
+            }
+
+            int i = 0;
+            if (Sse2.IsSupported)
+            {
+                fixed (int* leftPtr = m_array)
+                fixed (int* rightPtr = value.m_array)
+                {
+                    for (; i < count - (Vector128<int>.Count - 1); i += Vector128<int>.Count)
+                    {
+                        Vector128<int> leftVec = Sse2.LoadVector128(leftPtr + i);
+                        Vector128<int> rightVec = Sse2.LoadVector128(rightPtr + i);
+                        Sse2.Store(leftPtr + i, Sse2.And(leftVec, rightVec));
+                    }
+                }
             }
 
+            for (; i < count; i++)
+                m_array[i] &= value.m_array[i];
+
+_done:
```

Comment:
Nit: Coding conventions: "When using labels (for goto), indent the label one less than the current indentation."

Also, we typically use CamelCase for the label names.